framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -311,8 +423,140 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 		return false;
 	}
 
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	private async _createCCU() {}
+	private async _calculateInboxUpdate(
+		sendingChainID: Buffer,
+		chainAccount: ChainAccount,
+	): Promise<InboxUpdate> {
+		// TODO: Fetch as many CCMs as configured in _ccuFrequency.ccm, between lastCertificateHeight and certificateHeight - After #7569
+		const serialzedCCMs = this._chainConnectorState.crossChainMessages
+			.slice(0, this._ccuFrequency.ccm)
+			.map(ccm => codec.encode(ccmSchema, ccm));
+
+		const outboxKey = this._rawStateStoreKey(sendingChainID);
+
+		const stateProveResponse = await this._sidechainAPIClient.invoke<ProveResponse>('state_prove', {
+			stateRoot: chainAccount.lastCertificate.stateRoot,
+			queries: [outboxKey],
+		});
+		const inclusionProofOutboxRoot: OutboxRootWitness = {
+			bitmap: stateProveResponse.proof.queries[0].bitmap,
+			siblingHashes: stateProveResponse.proof.siblingHashes,
+		};
+
+		const inboxUpdate: InboxUpdate = {
+			crossChainMessages: serialzedCCMs,
+			messageWitnessHashes: [],
```

Comment:
```suggestion
			messageWitness: [],
```
shouldn't it be `messageWitness` like above? also when should this be filled?

Reply:
This comment is invalid since prop name is updated in the LIP